docassemble/assemblylinewizard/interview_generator.py

```diff
@@ -672,525 +717,43 @@ def delitem(self, *pargs):
-          if hasattr(self, 'id') and self.id:
-              content += "id: " + self.id + "\n"
-          else:
-              content += "id: " + oneline(self.question_text) + "\n"
-          if hasattr(self, 'event'):
-              content += "event: " + self.event + "\n"
-          content += "question: |\n"
-          content += indent_by(self.question_text, 2)
-          content += "subquestion: |\n"
-          content += indent_by(self.subquestion_text, 2)
-          content += "review: \n"
-          for parent_coll in self.parent_collections:
-            content += parent_coll.review_yaml() 
-            content += '  - note: |\n      ------\n'
-          for parent_coll in self.parent_collections:
-            content += parent_coll.revisit_page()
-            content += parent_coll.table_page()
-        return content
-
-class DAQuestionList(DAList):
-  """This represents a list of DAQuestions."""
-  def init(self, **kwargs):
-    super().init(**kwargs)
-    self.object_type = DAQuestion
-    # self.auto_gather = False
-    # self.gathered = True
-    # self.is_mandatory = False
+  Defaults to using the 'question' template and adds "field_list" attribute.
+  """
+  field_list:DAFieldList
+  
+  def init(self, *pargs, **kwargs):
+    super().init(*pargs, **kwargs)
+    self.template_key = 'question'
+    self.field_list = DAFieldList()
 
-  def all_fields_used(self):
-    """This method is used to help us iteratively build a list of fields that have already been assigned to a screen/question
-      in our wizarding process. It makes sure the fields aren't displayed to the wizard user on multiple screens.
-      It prevents the formatter of the wizard from putting the same fields on two different screens."""
-    fields = set()
-    for question in self.elements:
-      if hasattr(question,'field_list'):
-        for field in question.field_list.elements:
-          fields.add(field)
-    return fields
+  def source(self, template_string:str, imports:list=["from docassemble.assemblylinewizard.interview_generator import fix_id, varname, indent_by, mako_indent"])->str:
+    """
+    Return a string representing a YAML "document" (block), provided a string
+    representing a Mako template. Optional: provide list of imports.
+    """
+    mako.runtime.UNDEFINED = DAEmpty()
+    template = mako.template.Template(template_string, imports=imports)
+    data = {
+      "block_id": self.id if hasattr(self, 'id') else None,
+      "event": self.event if hasattr(self, 'event') else None,
+      "continue_button_field": varname(self.question_text) if self.needs_continue_button_field else None,
+      "question_text": self.question_text,
+      "subquestion_text": self.subquestion_text,
+      "field_list": self.field_list
+    }
+    return template.render(**data)
+
+def fix_id(string:str)->str:
+    if isinstance(string, str):
+      return re.sub(r'[\W_]+', ' ', string).strip()
+    else:
+      return ''     
```

Comment:
This is getting directly inserted into the mako templates now. 1) should this ever happen, 2) should we just throw if it does? Otherwise we know the output interview will have a ScannerError

Reply:
good point! I'll generate a UUID here instead. https://github.com/SuffolkLITLab/docassemble-assemblylinewizard/commit/2c151c2ed3c262a4188140d8ad152fec0a4c9628